plugin/pkg/admission/gc/gc_admission.go

```diff
@@ -61,11 +66,32 @@ func (a *gcPermissionsEnforcement) Admit(attributes admission.Attributes) (err e
 		Path:            "",
 	}
 	allowed, reason, err := a.authorizer.Authorize(deleteAttributes)
-	if allowed {
-		return nil
+	if !allowed {
+		return admission.NewForbidden(attributes, fmt.Errorf("cannot set an ownerRef on a resource you can't delete: %v, %v", reason, err))
+	}
+
+	// Further check if the user is setting ownerReference.blockOwnerDeletion.
+	// If so, only allows the change if the user has delete permission of
+	// the _OWNER_
+	changedRefs := changingBlockOwnerDeletion(attributes.GetObject(), attributes.GetOldObject())
+	for _, ref := range changedRefs {
+		attribute, err := a.toDeleteAttribute(ref, attributes)
+		if err != nil {
+			// An error occurs if a RESTMapping cannot be found for
```

Comment:
Good point. Further, if we are making the default "Forbidden" in the future, we should make it "Forbidden" now. Making validation rules tighter is backwards incompatible. Updated.